tests/kafkatest/services/kafka/kafka.py

```diff
@@ -277,6 +279,22 @@ def __init__(self, context, num_nodes, zk, security_protocol=SecurityConfig.PLAI
         self.isolated_controller_quorum = None # will define below if necessary
         self.configured_for_zk_migration = False
 
+        default_use_new_coordinator = False
+        
+        # If 'use_new_coordinator' is not explicitly set, determine it based on context.
+        if use_new_coordinator is None:
+            arg_name = 'use_new_coordinator'
+
+            # Default to the global setting if no arguments are injected.
+            if not context.injected_args:
```

Comment:
Are you sure about this change? What would happen if `context.injected_args` is defined but not for `use_new_coordinator`?

How about?

```
if use_new_coordinator is None:
   use_new_coordinator = context.injected_args.get(`use_new_coordinator`)
   if use_new_coordinator is None:
       use_new_coordinator = context.globals.get(`use_new_coordinator`)
```

Reply:
This is what I had before but this was giving a ERROR: `NoneType' object has no attribute 'get` on using get when use_new_coordinator isn't in the context args

Reply:
@imcdo Could you advice on how to get this right?

Reply:
How about?

```
if use_new_coordinator is None:
  use_new_coordinator = context.globals.get(`use_new_coordinator`)
  if context.injected_args is not None:
     use_new_coordinator = context.injected_args.get(`use_new_coordinator`, use_new_coordinator)
```

Reply:
> What would happen if context.injected_args is defined but not for use_new_coordinator

Reply:
I asked Ian in the office yesterday, he said the current implementation should work

Reply:
The code you provided would first take from globals and then from context, we want the first priority to be from context args and then global and if nothing is provided then we use the default value

Reply:
Hum... I suppose that I miss some understanding about how `injected_args` works. My understanding here is that whenever `injected_args` is defined, we take the value for `use_new_coordinator` from it or default to `default_use_new_coordinator`. This means that we basically ignore the global setting in this case.

What is the precedence of those here? value from the constructor, value from `injected_args` and value from `globals`?

Reply:
Injected args > globals > default where injected is in the args in each test, globals can be set at run time with ducktape arguments in the command line and default is the value we provided in this section of the code

Reply:
Right. But the code does not work like this, isn't it? if `injected_args` is present but `use_new_coordinator` is not in it, for whatever reason (say another feature), it ignores the globals, right?

Reply:
yep you're right, I changed it now. I misunderstood what you meant earlier.

Reply:
```
if use_new_coordinator is None:
  use_new_coordinator = context.globals.get(`use_new_coordinator`)
  if context.injected_args is not None:
     use_new_coordinator = context.injected_args.get(`use_new_coordinator`, use_new_coordinator)
```
This would've overwritten the global value with the default in case injected args didn't have the value specified. I changed it to something different now. This could've also allowed use_new_coordinator to be None in case the global value wasn't set and context args was None.